### PR TITLE
Update Hercules EPIC accounts

### PR DESCRIPTION
**Pull request by zach1221**
## Description

Updating the Hercules EPIC accounts to /work/noaa/epic/role-epic from role-epic-ps

### Input data additions/changes
- [ ] No changes are expected to input data.
- [ ] Changes are expected to input data:
  - [ ] New input data.
  - [ ] Updated input data.

### Anticipated changes to regression tests:
- [ ] No changes are expected to any regression test.
- [ ] Changes are expected to the following tests:

## Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [ ] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

## Commit Queue Checklist:

- [ ] Link PR's from all sub-components involved in section below
- [ ] Confirm reviews completed in ALL sub-component PR's
- [ ] Add all appropriate labels to this PR.
- [ ] Run full RT suite on either Hera/Cheyenne AND attach log to a PR comment.
- [ ] Add list of any failed regression tests to "Anticipated changes to regression tests" section.

## Testing Day Checklist:

- [ ] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.

### Testing Log (for CM's):
- RDHPCS
  - [ ] Hera
  - [ ] Orion
  - [ ] Jet
  - [ ] Gaea
  - [ ] Cheyenne
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- CI
  - [ ] Completed
- opnReqTest
  - [ ] N/A
  - [ ] Log attached to comment

### CODE REVIEW

**Comment by zach1221** [comment]
Alright, @ulmononian . I've seen you a review request for this new PR.

**Comment by ulmononian** [comment]
thanks for these updates, @zach1221. everything looks good, except that some paths to modulefiles and stacks do not yet exist (not your fault!). we have to get ecflow & mysql installed in the role-epic space before we actually put spack-stack/1.4.1 there (this is the latest version that results in all tests passing on hera). i'll try to get that done early this coming week and then, with just some minor changes to your pr, merge it in here.

File: modulefiles/ufs_hercules.gnu.lua [hunk 1]
```diff
@@ -2,7 +2,7 @@ help([[
 loads UFS Model prerequisites for Hercules/GNU
 ]])
 
-prepend_path("MODULEPATH", "/work/noaa/epic-ps/role-epic-ps/spack-stack/spack-stack-1.4.0-hercules/envs/unified-env-v2/install/modulefiles/Core")
+prepend_path("MODULEPATH", "/work/noaa/epic/role-epic/spack-stack/spack-stack-1.4.0-hercules/envs/unified-env-v2/install/modulefiles/Core")
```
Comment (ulmononian) [inline modulefiles/ufs_hercules.gnu.lua]: we should go with 1.4.1 here, but it does not yet exist on hera. see my comment in the PR conversation.

File: modulefiles/ufs_hercules.intel.lua [hunk 2]
```diff
@@ -2,7 +2,7 @@ help([[
 loads UFS Model prerequisites for Hercules/Intel
 ]])
 
-prepend_path("MODULEPATH", "/work/noaa/epic-ps/role-epic-ps/spack-stack/spack-stack-1.4.0-hercules/envs/unified-env-v2/install/modulefiles/Core")
+prepend_path("MODULEPATH", "/work/noaa/epic/role-epic/spack-stack/spack-stack-1.4.0-hercules/envs/unified-env-v2/install/modulefiles/Core")
```
Comment (ulmononian) [inline modulefiles/ufs_hercules.intel.lua]: same as above.

File: tests/rt.sh [hunk 3]
```diff
@@ -265,15 +265,15 @@ elif [[ $MACHINE_ID = hercules ]]; then
   export PATH=/apps/spack-managed/gcc-11.3.1/miniconda3-4.10.3-un3f2xdus7rbrzgso5ketsq4gp2iociv/bin:$PATH
   export PYTHONPATH=/apps/spack-managed/gcc-11.3.1/miniconda3-4.10.3-un3f2xdus7rbrzgso5ketsq4gp2iociv/lib/python3.9/site-packages
 
-  module use /work/noaa/epic-ps/role-epic-ps/spack-stack/modulefiles
+  module use /work/noaa/epic/role-epic/spack-stack/modulefiles
```
Comment (ulmononian) [inline tests/rt.sh]: this modulefiles dir. and the packages that the modules will point to have yet to be installed on hercules in this new space (reiterating my comment from the pr conversation page). but this is how it should look once those are in place!

File: modulefiles/ufs_hercules.gnu.lua (hunk 1 above)
Comment (zach1221) [reply modulefiles/ufs_hercules.gnu.lua]: This has been updated

File: modulefiles/ufs_hercules.intel.lua (hunk 2 above)
Comment (zach1221) [reply modulefiles/ufs_hercules.intel.lua]: This has been updated also

File: tests/rt.sh (hunk 3 above)
Comment (zach1221) [reply tests/rt.sh]: Thanks @ulmononian . I'll wait until ecflow, mysql and spack-stack/1.4.1 are installed in the role-epic spaces. Then we can revisit!

Comment (ulmononian) [reply tests/rt.sh]: @zach1221 ecflow & mysql have been installed. the updated modulefile path is `/work/noaa/epic/role-epic/spack-stack/hercules/modulefiles`.

File: tests/rt.sh [hunk 4]
```diff
@@ -265,15 +265,15 @@ elif [[ $MACHINE_ID = hercules ]]; then
...
   module load ecflow/5.8.4-hercules
-  ECFLOW_START=/work/noaa/epic-ps/role-epic-ps/spack-stack/ecflow-5.8.4-hercules/bin/ecflow_start.sh
+  ECFLOW_START=/work/noaa/epic/role-epic/spack-stack/ecflow-5.8.4-hercules/bin/ecflow_start.sh
```
Comment (ulmononian) [inline tests/rt.sh]: `ECFLOW_START` path should be changed to:

`/work/noaa/epic/role-epic/spack-stack/hercules/ecflow-5.8.4/bin/ecflow_start.sh`

File: tests/rt.sh [hunk 5]
```diff
@@ -265,15 +265,15 @@ elif [[ $MACHINE_ID = hercules ]]; then
   export PATH=/apps/spack-managed/gcc-11.3.1/miniconda3-4.10.3-un3f2xdus7rbrzgso5ketsq4gp2iociv/bin:$PATH
   export PYTHONPATH=/apps/spack-managed/gcc-11.3.1/miniconda3-4.10.3-un3f2xdus7rbrzgso5ketsq4gp2iociv/lib/python3.9/site-packages
 
-  module use /work/noaa/epic-ps/role-epic-ps/spack-stack/modulefiles
+  module use /work/noaa/epic/role-epic/spack-stack/modulefiles
   module load ecflow/5.8.4-hercules
```
Comment (ulmononian) [inline tests/rt.sh]: this should be changed to 

`module load ecflow/5.8.4`

File: tests/rt.sh (hunk 3 above)
Comment (zach1221) [reply tests/rt.sh]: updated

File: tests/rt.sh (hunk 4 above)
Comment (zach1221) [reply tests/rt.sh]: updated

File: tests/rt.sh (hunk 5 above)
Comment (zach1221) [reply tests/rt.sh]: done

File: modulefiles/ufs_hercules.intel.lua [hunk 6]
```diff
@@ -2,7 +2,7 @@ help([[
 loads UFS Model prerequisites for Hercules/Intel
 ]])
 
-prepend_path("MODULEPATH", "/work/noaa/epic-ps/role-epic-ps/spack-stack/spack-stack-1.4.0-hercules/envs/unified-env-v2/install/modulefiles/Core")
+prepend_path("MODULEPATH", "/work/noaa/epic/role-epic/spack-stack/spack-stack-1.4.1-hercules/envs/unified-env-v2/install/modulefiles/Core")
```
Comment (ulmononian) [inline modulefiles/ufs_hercules.intel.lua]: sorry...one more:

there are issues with gcc 12 on hercules, so neither the intel or gnu stacks are completed yet, but the path will be:

`/work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.4.1/envs/unified-env/install/modulefiles/Core`

File: modulefiles/ufs_hercules.gnu.lua [hunk 7]
```diff
@@ -2,7 +2,7 @@ help([[
 loads UFS Model prerequisites for Hercules/GNU
 ]])
 
-prepend_path("MODULEPATH", "/work/noaa/epic-ps/role-epic-ps/spack-stack/spack-stack-1.4.0-hercules/envs/unified-env-v2/install/modulefiles/Core")
+prepend_path("MODULEPATH", "/work/noaa/epic/role-epic/spack-stack/spack-stack-1.4.1-hercules/envs/unified-env-v2/install/modulefiles/Core")
```
Comment (ulmononian) [inline modulefiles/ufs_hercules.gnu.lua]: same as my comment for the intel modulefile.

File: modulefiles/ufs_hercules.intel.lua (hunk 6 above)
Comment (zach1221) [reply modulefiles/ufs_hercules.intel.lua]: done

File: modulefiles/ufs_hercules.gnu.lua (hunk 7 above)
Comment (zach1221) [reply modulefiles/ufs_hercules.gnu.lua]: done

**Review by ulmononian — APPROVED** [review]
sorry for all the iterations.

looks great!

will merge now.

**Pull request merged**